packager/mpd/base/mpd_utils.cc

```diff
@@ -162,6 +162,15 @@ std::string GetAdaptationSetKey(const MediaInfo& media_info,
   if (!ignore_codec) {
     key.append(":");
     key.append(GetBaseCodec(media_info));
+
+    if (GetBaseCodec(media_info).find("dvh") == 0) {
```

Comment:
@sr1990  could you explain the logic here a bit ? 
what's this conditional logic for dvh for ?

Reply:
@vish91 This is similar to https://github.com/shaka-project/shaka-packager/blob/80e024013df87a4bfeb265c8ea83cfa2a0c5db0f/packager/hls/base/media_playlist.cc#L554
which is used to get HLS VIDEO-RANGE.
VUI which contains the transfer characteritics is an optional section of syntax within the SPS.
As per spec at https://www.itu.int/rec/T-REC-H.265-202108-I/en
```
vui_parameters_present_flag equal to 1 specifies that the vui_parameters( ) syntax structure as specified in Annex E is
present. vui_parameters_present_flag equal to 0 specifies that the vui_parameters( ) syntax structure as specified in
Annex E is not present.
```

This check is done while parsing SPS
H265Parser::ParseSps https://github.com/shaka-project/shaka-packager/blob/80e024013df87a4bfeb265c8ea83cfa2a0c5db0f/packager/media/codecs/h265_parser.cc#L628
and H264Parser::ParseSps
https://github.com/shaka-project/shaka-packager/blob/80e024013df87a4bfeb265c8ea83cfa2a0c5db0f/packager/media/codecs/h264_parser.cc#L695

In case of dolby vision content, even if the VUI is not present, the transfer characteristics are set to PQ.
I will go ahead and add
`  // Dolby Vision (dvh1 or dvhe) is always HDR.`
comment which is also present in MediaPlaylist::GetVideoRange().

Reply:
@vish91, does that satisfy your concern?

Reply:
yeah this makes sense.  :shipit: